README.md

```diff
@@ -106,6 +106,160 @@ of each other.
+
+### How can I change what's in the alert?
+
+You can use the field ``alert_text`` to add custom text to an alert. By setting ``alert_text_type: alert_text_only``, it will be the entirety of the alert. You can also add different fields from the alert by using Python style string formatting and ``alert_text_args``. For example
+
+```
+alert_text: "Something happened with {0} at {1}"
+alert_text_type: alert_text_only
+alert_text_args: ["username", "@timestamp"]
+```
+
+You can also limit the alert to only containing certain fields from the document by using ``include``.
+
+```
+include: ["ip_address", "hostname", "status"]
+```
+
+### My alert only contains data for one event, how can I see more?
+
+If you are using ``type: frequency``, you can set the option ``attach_related: true`` and every document will be included in the alert. An alternative, which works for every type, is ``top_count_keys``. This will show the top counts for each value for certain fields. For example, if you have
+
+```
+top_count_keys: ["ip_address", "status"]
+```
+
+and 10 documents matched your alert, it may contain something like
+
+```
+ip_address:
+127.0.0.1: 7
+10.0.0.1: 2
+192.168.0.1: 1
+
+status:
+200: 9
+500: 1
+```
+
+### How can I make the alert come at a certain time?
+
+The ``aggregation`` feature will take every alert that has occured over a period of time and send them together in one alert. You can use cron style syntax to send all alerts that have occured since the last once by using
+
+```
+aggregation:
+  schedule: '2 4 * * mon,fri'
+```
+
+### I have lots of documents and it's really slow, how can I speed it up?
+
+There are several ways to potentially speed up queries. If you are using ``index: logstash-*``, Elasticsearch will query all shards, even if they do not possibly contain data with the correct timestamp. Instead, you can use Python time format strings and set ``use_strftime_index``
+
+```
+index: logstash-%Y.%m
+use_strftime_index: true
+```
+
+Another thing you could change is ``buffer_time``. By default, ElastAlert will query large overlapping windows in order to ensure that it does not miss any events, even if they are indexed in real time. In config.yaml, you can adjust ``buffer_time`` to a smaller number to only query the most recent few minutes.
+
+```
+buffer_time:
+  minutes: 5
+```
+
+By default, ElastAlert will download every document in full before making processing them. Instead, you can have ElastAlert simply get a count of the number of documents that have occured in between each query. To do this, set ``use_count_query: true``. This cannot be used if you use ``query_key``, because ElastAlert will not know the contents of each documents, just the total number of them. This also reduces the precision of alerts, because all documents that occur between each query will be rounded to a single timestamp.
```

Comment:
1. "before making processing" --> "before processing"
2. "all documents that occur" --> "all events that occur"